src/lib/memory/blob_manager.cpp

```diff
@@ -3,33 +3,43 @@
 namespace homeobject {
 
 BlobManager::Result< blob_id > MemoryHomeObject::_put_blob(ShardInfo const& shard, Blob&& blob) {
-    auto lg = std::scoped_lock(_index_lock);
-    auto [btree_it, h1] = _in_memory_index.try_emplace(shard.id, std::make_pair(btree(), blob_id(0ull)));
-    RELEASE_ASSERT(_in_memory_index.end() != btree_it, "Could not create BTree!");
-
-    auto route = BlobRoute{shard.id, btree_it->second.second++};
-    LOGDEBUGMOD(homeobject, "Writing BLOB {} to: BlkId:[{}]", route.blob, fmt::ptr(&blob));
+    auto route = BlobRoute{shard.id, 0ull};
+    auto index_it = _in_memory_index.end();
+    {
+        auto lg = std::scoped_lock(_index_lock);
```

Comment:
It will better simulate the behavior (fail put_blob if shard doestn exists) if we put  shardIndex creation into `_create_shard` 
so that. in `_put_blob` we only take `shared_lock` and return an error if shard doestn exists

Reply:
Let me try that; one moment :+1:

Reply:
also if we can check `sealed` flag